Automattic/jetpack
# Help Center: extend post fetch endpoint

## Proposed changes:

Adds the `post_url` arg to `help-center/fetch-post` endpoint.

This is part of a larger task to allow support guide links in Help Center open within Help Center.
PR for the main package ⇢ https://github.com/Automattic/wp-calypso/pull/92745
PR for the main endpoint ⇢ D155970-code

### Other information:


## Does this pull request change what data or activity we track or use?

Nope

## Testing instructions:

This does not add any feature to test so really we are only testing to make sure the endpoint works and that Help Center does not break.

1. Open the dev console in your browser.
2. Open the Help Center and open a support guide.
3. Search for a guide and open that as well.
4. Go to the Network tab in your dev console and view the request for `/fetch-post`
5. Copy this request as a `fetch`
6. Paste this in the console and replace `post_id` and `blog_id` with `post_url` and the url of an article.
7. All requests should return 200 with the correct article.

```
fetch("https://busineselini.wpcomstaging.com/wp-json/help-center/fetch-post?post_id=158974&blog_id=9619154&_locale=user", {
"headers": ...
  });
```

```
fetch("https://busineselini.wpcomstaging.com/wp-json/help-center/fetch-post?post_url=https://en.support.wordpress.com/getting-started-with-wordpress-com/&_locale=user", {
  "headers": ...
  });
```
